site/content/concepts/tenancy.md

```diff
@@ -1,12 +1,149 @@
+
+But if necessary, this can be extended:  
+
+**Advanced setup for multiple tenants**
+
+Due to the logical view Hono has to devices, a physical device can have several *logical identities* in Hono: if using
+different credentials when connecting to a protocol adapter, the physical device can *decide* which identity to use
+for this connection, and thus can even connect in the context of different tenants.
+
+
+## Upstream data flow for tenants
+
+An important detail in Hono's architecture is that data sent upstream is transported via the tenant
+scoped AMQP 1.0 links from the protocol adapters to the AMQP 1.0 network.
+
+Each tenant has its own pair of AMQP 1.0 links and is treated 
+independently from other tenants regarding the back pressure mechanism that AMQP 1.0 offers (using different credits).
+
+This addresses the noisy neighbour problem that needs to be solved in a multitenant setup.  
+ 
+## Microservice authorization for tenants
+
+Hono's components authenticate each other by means of the [Authentication API]({{< relref "Authentication-API.md" >}}).
+
+The returned token for a successful authentication contains authorization information that is addressing the AMQP 1.0
+endpoints. Since the endpoints (as outlined above) are scoped to a tenant, this enables to configure tenants that are
+authorized to only a subset of Hono's full functionality.
+
+Example:
+
+Tenants can be defined by means of authorization configuration that are only allowed to send telemetry data upstream,
+but not event data. This can be done by not including the event endpoint in the authorization token for these tenants.
+
+## Business applications and tenants
+
+The northbound *Business applications* are always connecting to the AMQP 1.0 endpoints of Hono.
+ 
+By means of the authentication and authorization setup and the fact that the endpoints are scoped to a tenant, the 
+*Business application* is only acting in the context of one tenant.
+
+**Advanced setup for multiple tenants**
+
+If the authentication and authorization setup includes endpoints of more than one tenant, this also allows *Business applications*
+to access data of multiple tenants.
+
+## Separation of tenants
+
+Tenants are separated from each other in all of Hono's components. 
+
+Here is a summary of how this is implemented:
+
+- the registration of devices are strictly scoped to a tenant
+- the credentials of devices are strictly scoped to a tenant
+- protocol adapters can be enabled/disabled for a tenant 
+- the upstream data flow is isolated for every tenant
+- *Business applications* need to authenticate to the AMQP 1.0 network and are by that mechanism scoped to their tenant
+ 
+## Notes on a production ready setup
+
+To be flexible for the different needs of production setups, Hono tries to make as few assumptions about the combination
+of the different APIs as possible.
+This means e.g. that there is no connection between the example implementations of the APIs:
```

Comment:
Maybe we can replace *connection* with *referential integrity being enforced*?

I also think that we should more explicitly point out that the *Device Registry* component does not do this, but that a production system would (should) certainly enforce this. The term *example implementations* is a little abstract, when in fact we talk about Hono's *Device Registry* component, don't you think?